rotkehlchen/chain/bitcoin/manager.py

```diff
@@ -183,66 +170,155 @@ def _query_blockchain_info_has_transactions(
+                    handle_429=True,
+                    backoff_in_seconds=10,
+                )['txs'])
+                if len(chunk) < limit:
+                    break  # all txs have been queried
+
+                offset += limit
+
+        return self._process_accounts_raw_tx_lists(
+            accounts_raw_tx_lists=results,
+            options=options,
+            processing_fn=BitcoinTx.deserialize_from_blockchain_info,
+        )
 
-        return tx_list
+    def _process_raw_tx_from_blockcypher(
+            self,
+            data: dict[str, Any],
+    ) -> BitcoinTx:
+        """Convert a raw tx dict from blockcypher into a BitcoinTx.
+        If the tx has a large number of TxIOs, the remaining TxIOs will be queried using the urls
+        provided in the API response.
+        May raise DeserializationError, KeyError, RemoteError, UnableToDecryptRemoteData.
+        """
+        inputs: list[dict[str, Any]] = []
+        outputs: list[dict[str, Any]] = []
+        for side, tx_io_list in (('inputs', inputs), ('outputs', outputs)):
+            next_data = data.copy()
+            while True:
+                tx_io_list.extend(list_chunk := next_data[side])
+                if (
+                    (next_url := next_data.get(f'next_{side}')) is None or
+                    len(list_chunk) < BLOCKCYPHER_TX_IO_LIMIT
+                ):
+                    break  # all TxIOs for this side have been queried
+
+                next_data = request_get_dict(url=next_url, handle_429=True, backoff_in_seconds=1)
+
+        data['inputs'] = inputs
+        data['outputs'] = outputs
+        return BitcoinTx.deserialize_from_blockcypher(data)
+
+    def _query_blockcypher_transactions(
+            self,
+            accounts: Sequence[BTCAddress],
+            options: dict[str, Any],
+    ) -> tuple[int, list[BitcoinTx]]:
+        """Query blockcypher for transactions.
+        Txs from the api are ordered newest to oldest, with pagination via block_height.
+        Returns a tuple containing the latest queried block height and the list of txs.
+        """
+        accounts_tx_lists: dict[BTCAddress, list[dict[str, Any]]] = defaultdict(list)
+        limits = f'limit={BLOCKCYPHER_TX_LIMIT}&txlimit={BLOCKCYPHER_TX_IO_LIMIT}'
+        for accounts_chunk in get_chunks(list(accounts), BLOCKCYPHER_BATCH_SIZE):
+            before_height = None
+            while len(accounts_chunk) > 0:
+                url = f"{BLOCKCYPHER_BASE_URL}/addrs/{';'.join(accounts_chunk)}/full?{limits}"
+                if before_height is not None:
+                    url += f'&before={before_height}'
+
+                for entry in [response] if isinstance((response := request_get(
+                    url=url,
+                    handle_429=True,
+                    backoff_in_seconds=1,  # the free rate limit is 3 requests per second
+                )), dict) else response:  # dict/list depending on single/multiple accounts
```

Comment:
This is **very**, **very** hard to read. I know we prefer using `:=` almost everywhere but here it just makes it hard to understand
```suggestion
                response := request_get(
                    url=url,
                    handle_429=True,
                    backoff_in_seconds=1,  # the free rate limit is 3 requests per second
                )
                for entry in [response] if isinstance(response, dict) else response:  # dict/list depending on single/multiple accounts  # noqa: E501
```